### PR TITLE
Added a function for calculating the intercept

### PR DESCRIPTION
**Pull request by horusscope**
e.g. for a top-down view RTS game
It seemed very improper to even consider collisions in this context.

### CODE REVIEW

File: src/Culling/ray.ts [hunk 1]
```diff
@@ -245,6 +245,24 @@ export class Ray {
             return distance;
         }
     }
+    /**
+     * Calculate the intercept of a ray on a given axis
+     * @param axis to check 'x' | 'y' | 'z'
+     * @returns a vector containing the coordinates where 'axis' is equal to zero, or false if there is no intercept.
+     */
+    public intersectsAxis(axis) {
```
Comment (deltakosh) [inline src/Culling/ray.ts]: axis should be typed as string

File: src/Culling/ray.ts [hunk 2]
```diff
@@ -245,6 +245,24 @@ export class Ray {
...
     }
+    /**
+     * Calculate the intercept of a ray on a given axis
+     * @param axis to check 'x' | 'y' | 'z'
+     * @returns a vector containing the coordinates where 'axis' is equal to zero, or false if there is no intercept.
+     */
+    public intersectsAxis(axis) {
+          var t = this.origin[axis] / this.direction[axis]
+          if(t > 0) return false
+          const stepOrigin = function(t, axis) {
```
Comment (deltakosh) [inline src/Culling/ray.ts]: for performance and GC reason we should not instantiate a function every time we call intersectsAxis

File: src/Culling/ray.ts (hunk 1 above)
Comment (horusscope) [reply src/Culling/ray.ts]: sorry about that, I only liked the readability.

**Comment by deltakosh** [comment]
There is a build issue :)

**Pull request merged**